lib/src/scenario_generator.dart

```diff
@@ -16,10 +17,18 @@ void parseScenario(
   sb.writeln(
     "    $testMethodName('''$scenarioTitle''', ($testerName) async {",
   );
-  if (hasTearDown) {
+  if (hasHooks) {
+    sb.writeln('      var $testSuccessVariableName = true;');
+  }
+  if (hasTearDown || hasHooks) {
     sb.writeln('      try {');
   }
   final spaces = hasTearDown ? '        ' : '      ';
+  if (hasHooks) {
+    sb.writeln(
+      '${spaces}await $setUpHookName("$scenarioTitle" ${tags.isNotEmpty ? ', ${tagsToString(tags)}' : ''});',
```

Comment:
It is unsafe assumption that users aren't using quotes in scenario titles, please use `'''` here and in other places.
```suggestion
      '${spaces}await $setUpHookName('''$scenarioTitle''' ${tags.isNotEmpty ? ', ${tagsToString(tags)}' : ''});',
```